dev-libs/crypto++/crypto++-5.6.5.recipe

```diff
@@ -14,28 +14,33 @@ The library contains the following algorithms:
 * Elliptic curve cryptography: ECDSA, ECNR, ECIES, ECDH, ECMQV
 * Insecure or obsolescent algorithms: MD2, MD4, MD5
 * And many more other algorithms."
-HOMEPAGE="https://cryptopp.com"
+HOMEPAGE="https://cryptopp.com/"
 COPYRIGHT="2015-2017 Wei Dai"
 LICENSE="Boost v1.0"
-REVISION="2"
+REVISION="3"
 SOURCE_URI="https://www.cryptopp.com/cryptopp565.zip"
 CHECKSUM_SHA256="a75ef486fe3128008bbb201efee3dcdcffbe791120952910883b26337ec32c34"
 SOURCE_DIR=""
+# Upstream patch, might be obsolete in a later version
+PATCHES="crypto++-$portVersion.patchset"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersion=$portVersion
+libVersionCompat=${libVersion%.*}
```

Comment:
Hmm, libVersionCompat is already in 4 recipes:
* dev-libs/libpipeline/libpipeline-1.5.0.recipe
* net-libs/libblkmaker/libblkmaker-0.5.3.recipe
* sci-mathematics/gmp_ecm/gmp_ecm-7.0.4.recipe
* sys-libs/hwloc/hwloc-1.11.8.recipe

and it is defined differently.
I though about changing these 4 recipes to adjust to what we have in this PR right now, but the consequence would be that we would have much longer lines.
So I think it would be better to only change crypto++ this way:
1) first we rename to libVersionShort what we were calling libVersionCompat.
2) then we define libVersionCompat as `$libVersion compat >= $libVersionShort`

These changes would allow to have a consistent definition of libVersionCompat across the 4 (and soon 5) recipes which use it. I cannot see any better alternative...
What do you think of "libVersionShort" ? I first though naming this libVersionLink, then libVersionSymLink, but it looks like libVersionShort is the less confusing. What do you think?
Thanks!

```diff
 libVersion=$portVersion
-libVersionCompat=${libVersion%.*}
+libVersionShort=${libVersion%.*}
+libVersionCompat="$libVersion compat >= $libVersionShort"
```

Reply:
The name doesn't actually make any sense. It's not the "short" version number, nor the version we symlinked, but rather the ABI version that's supported by that version of the library.

I suggest we drop `libVersionShort` and do this

```diff
 libVersion=$portVersion
-libVersionCompat=${libVersion%.*}
+libVersionCompat="$libVersion compat >= ${libVersion%.*}"
```

And for the symlink

```diff
-	ln -s libcryptopp.so.$libVersion $libDir/libcryptopp.so.$libVersionCompat
+	ln -s libcryptopp.so.$libVersion $libDir/libcryptopp.so.${libVersionCompat##* }
```

For the other ports, I suggest using `${libVersion%.*}` or `${libVersion%%.*}` depending on the authors versioning scheme. This will ensure that maintainers would only need to change one line when `SONAME` bump happens instead of two, reducing the margin of error. And IMO, the less constants we depend on the better, especially when it comes to versioning.

What's your opinion on this?

Reply:
Excellent idea! You rock, @alaviss !
To summarize, we would use:

* for libs which have a `libfoobar.so.x -> libfoobar.x.y.z`:
```
libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
```

* for libs which have a `libfoobar.so.x.y -> libfoobar.x.y.z`:
```
libVersionCompat="$libVersion compat >= ${libVersion%.*}"
```

and we would set libVersion to either portVersion, if this is appropriate, or to a the appropriate value if it is different from the portVersion.

And since there are very few recipes which need to compute `${libVersionCompat##* }`, this is not a problem.

So OK for me!